### PR TITLE
Mike | refactor transformations, add some responsive styling to table…

### PR DESCRIPTION
**Pull request by mike-robertson**
… so it is useable in mobile

### CODE REVIEW

File: transformations/playerMatchTransform.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
-  if (transformation[field])
-  {
-    return transformation[field](match, match[field], constants);
-  }
-  else
-  {
-    return match[field];
+
+const playerMatchTransform = (match, field, constants) => {
+  switch (field) {
```
Comment (howardchung) [inline transformations/playerMatchTransform.js]: Why was this changed?

File: transformations/transformations.js [hunk 2]
```diff
@@ -1,16 +1,12 @@
...
-  return p.player_slot < 128;
-}
+// TODO move to utility module
+const isRadiant = (p) => p.player_slot < 128;
 
-function pad(n, width, z) {
-    z = z || '0';
-    n = n + '';
-    return n.length >= width ? n : new Array(width - n.length + 1).join(z) + n;
+function pad(n, width, z = '0') {
```
Comment (howardchung) [inline transformations/transformations.js]: This was ported from existing code.  Be absolutely sure it works if you want to refactor

File: transformations/transformations.js [hunk 3]
```diff
@@ -22,20 +18,20 @@ function format(input) {
...
+  return time;
 }
 
 const transformation = {
-  hero_id: (m, v, c) => `${HOST_URL}${c.heroes[v] ? c.heroes[v].img: v}`,
-  radiant_win: (m, v, c) => isRadiant(m) === v ? 'W' : 'L',
-  game_mode: (m, v, c) => (c.game_mode[v] ? c.game_mode[v].name : v),
-  start_time: (m, v, c) => moment(v, 'X').fromNow(),
-  duration: (m, v, c) => formatSeconds(v),
+  heroId: (field, { heroes }) => `${HOST_URL}${heroes[field] ? heroes[field].img : field}`,
```
Comment (howardchung) [inline transformations/transformations.js]: Rename m,v,c but don't change otherwise

File: transformations/transformations.js (hunk 2 above)
Comment (mike-robertson) [inline transformations/transformations.js]: ES lint failure

7.12 Never mutate parameters. eslint: no-param-reassign

File: transformations/playerMatchTransform.js (hunk 1 above)
Comment (mike-robertson) [inline transformations/playerMatchTransform.js]: see discord

File: transformations/transformations.js (hunk 3 above)
Comment (mike-robertson) [inline transformations/transformations.js]: see discord
